### ISSUE TITLE
Cannot edit a preexisting note file (Calcurse 4.0.0)

### ISSUE THREAD

**Issue by chri87d9** [lfos/calcurse#279]
**Bug description.** Provide a clear and concise description of what the bug is.
When a note filename is manually placed in the apts file, it copys the contents to a new file and saves it there.

Example:
I am trying to create a logbook system for satellite passes. My scrip inserts the passes into the apts file, and creates md templates for the log. I then link the appointment to the file in the /notes/ dir. Thereby it would be easy to navigate passes and keep track of logs.

**Comment by lfos** [comment]
The file name of notes has to be the SHA1 hash of the note itself. That's by design...

**Comment by chri87d9** [comment]
Is there a particular reason for why both cannot coexist? Meaning that if a filename is manually put in, and the file is there it will work properly. And if not, it will generate the file in the usual way?

**Comment by lfos** [comment]
calcurse does not store the file name of the note internally, therefore it doesn't remember what your manual entry was when saving the data files. Is there a reason you cannot change your script to use the SHA1 hash of the note as file name?

**Comment by chri87d9** [comment]
No, not a such. But it is more that it would be easier to write script for interfacing with the files and navigating them outside calcurse. I was intending on writing scripts that could maybe take the note files (md, groff or tex) and merge and compile them. It would therefore be nice to have a way of ordering the files correctly, and maybe selecting specific date ranges to compile.

But it has turned out to be more of a feature request than a bug/issue.

**Comment by lfos** [comment]
Since we use the SHA1 of the note text, two items with exactly the same note will share the same note file. Thus, there's no meaningful way of ordering notes by date.

**Comment by lfos** [comment]
That being said, there has been the idea of supporting named notes, which could be implemented as symlinks to the actual (SHA1-named) note files. However, you still would not be able to edit notes without updaing their file names.
